marklogic-spark-connector/src/main/java/com/marklogic/spark/writer/DocBuilder.java

```diff
@@ -156,8 +160,12 @@ private DocumentWriteOperation buildMainDocument(DocumentInputs inputs) {
 
         AbstractWriteHandle content = inputs.getContent();
         byte[] classificationResponse = inputs.getClassificationResponse();
-        if (classificationResponse != null && inputs.getExtractedText() == null && Format.XML.equals(((BytesHandle) content).getFormat())) {
-            content = appendDocumentClassificationToXmlContent(((BytesHandle) content).get(), inputs.getInitialUri(), classificationResponse);
+        if (classificationResponse != null && inputs.getExtractedText() == null) {
+            if (Format.XML.equals(((BytesHandle) content).getFormat())) {
+                content = appendDocumentClassificationToXmlContent(((BytesHandle) content).get(), inputs.getInitialUri(), classificationResponse);
+            } else if (Format.JSON.equals(((BytesHandle) content).getFormat())) {
+                content = appendDocumentClassificationToJsonContent(((BytesHandle) content).get(), inputs.getInitialUri(), classificationResponse);
```

Comment:
The Util class in the langchain4j subproject also has a `getJsonFromHandle` method - use that here instead of casting to BytesHandle. It also handles that annoying checked exception from Jackson.